app/src/androidTest/java/org/fossasia/susi/ai/helper/ValidateUtilsTest.kt

```diff
@@ -0,0 +1,42 @@
+package org.fossasia.susi.ai.helper
+
+import junit.framework.Assert.assertFalse
+import junit.framework.Assert.assertTrue
+import org.junit.Test
+
+class ValidateUtilsTest {
+
+    @Test
+    fun testValidateUrl() {
+        assertTrue(ValidateUtils.validateUrl("https://www.test.com"))
+        assertTrue(ValidateUtils.validateUrl("http://www.test.com"))
+        assertTrue(ValidateUtils.validateUrl("ftp://www.test.com"))
+        assertTrue(ValidateUtils.validateUrl("file://www.test.com"))
+        assertTrue(ValidateUtils.validateUrl("https://test.com"))
+        assertTrue(ValidateUtils.validateUrl("https://www.TEST.co"))
+        assertTrue(ValidateUtils.validateUrl("https://test"))
+        assertTrue(ValidateUtils.validateUrl("https://www.TEST.c@!34o"))
+        assertTrue(ValidateUtils.validateUrl("https://www.test5465uy.co.in"))
+        assertTrue(ValidateUtils.validateUrl("https://www.te/**-st.co.in"))
+        assertTrue(ValidateUtils.validateUrl("https://www.test.co.in"))
+        assertTrue(ValidateUtils.validateUrl("https://www.test.co"))
+        assertTrue(ValidateUtils.validateUrl("https://www.test.in"))
+        assertTrue(ValidateUtils.validateUrl("https://54652"))
+        assertTrue(ValidateUtils.validateUrl("https://t.co"))
+        assertTrue(ValidateUtils.validateUrl("https://tby.test.com"))
+
+        assertFalse(ValidateUtils.validateUrl(""))
+        assertFalse(ValidateUtils.validateUrl("ht://www.test.com"))
+        assertFalse(ValidateUtils.validateUrl("http/www.test.com"))
+        assertFalse(ValidateUtils.validateUrl("http://"))
+        assertFalse(ValidateUtils.validateUrl("http://."))
+        assertFalse(ValidateUtils.validateUrl("http://...."))
+        assertFalse(ValidateUtils.validateUrl("http56://www.test.com"))
+        assertFalse(ValidateUtils.validateUrl("http//www.test."))
+        assertFalse(ValidateUtils.validateUrl("http//www."))
+        assertFalse(ValidateUtils.validateUrl("httpstest.com"))
+        assertFalse(ValidateUtils.validateUrl("test"))
+        assertFalse(ValidateUtils.validateUrl("test.co.in"))
+        assertFalse(ValidateUtils.validateUrl(".com"))
+    }
```

Comment:
Useless tests. You should test the method I commented on and not a new method you created which isn't used anywhere

Reply:
Do I need to mock the skildata object?

Reply:
No, you need to create it

Reply:
@iamareebjamal I am unable to understand how to create the test, could you help a little ?

Reply:
@arundhati24 it would be great if you could also help here a bit, as a new release has to be after this on both FDroid and Playstore

Reply:
Pass in SkillData object and verify the URL which will return from the function